AndroidSDKCore/src/main/java/com/leanplum/internal/LeanplumEventDataManager.java

```diff
@@ -48,53 +48,58 @@
  * @author Anna Orlova
  */
 public class LeanplumEventDataManager {
+
+  private static LeanplumEventDataManager instance;
+
   private static final String DATABASE_NAME = "__leanplum.db";
   private static final int DATABASE_VERSION = 1;
   private static final String EVENT_TABLE_NAME = "event";
   private static final String COLUMN_DATA = "data";
   private static final String KEY_ROWID = "rowid";
 
-  private static SQLiteDatabase database;
-  private static LeanplumDataBaseManager databaseManager;
-  private static ContentValues contentValues = new ContentValues();
+  private SQLiteDatabase database;
+  private LeanplumDataBaseManager databaseManager;
+  private ContentValues contentValues = new ContentValues();
+  private boolean sendErrorLogs = false;
 
-  static boolean willSendErrorLog = false;
+  private LeanplumEventDataManager() {
+    try {
+      Context context = Leanplum.getContext();
 
-  /**
-   * Creates connection to database, if database is not present, it will automatically create it.
-   *
-   * @param context Current context.
-   */
-  public static void init(Context context) {
-    if (database != null) {
-      Log.e("Database is already initialized.");
-      return;
-    }
+      if (context == null) {
+        return;
+      }
 
-    // Create database if needed.
-    try {
       if (databaseManager == null) {
-        databaseManager = new LeanplumDataBaseManager(context);
+        databaseManager = new LeanplumDataBaseManager(Leanplum.getContext());
       }
       database = databaseManager.getWritableDatabase();
+
     } catch (Throwable t) {
       handleSQLiteError("Cannot create database.", t);
     }
   }
 
+  public static LeanplumEventDataManager sharedInstance() {
+    if (instance == null) {
+      instance = new LeanplumEventDataManager();
+    }
+    return instance;
+  }
+
   /**
    * Inserts event to event table.
    *
    * @param event String with json of event.
    */
-  static void insertEvent(String event) {
```

Comment:
To confirm here. So since LeanplumEventDataManager is singleton, you dont need static?

Reply:
Yeah, its singleton now and no need for static methods.